### PR TITLE
kraken: An OSD was seen getting ENOSPC even with osd_failsafe_full_ratio passed

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/19340
http://tracker.ceph.com/issues/19841
http://tracker.ceph.com/issues/20672

### CODE REVIEW

**Comment by smithfarm** [comment]
@liewegas @jdurgin @dzafman This passed a rados suite at http://tracker.ceph.com/issues/19009#note-62

Please review.

**Comment by dzafman** [comment]
Please add #16220 to this backport which fixes a problem with 489e810

File: src/mon/PGMonitor.cc [hunk 1]
```diff
@@ -1796,6 +1796,32 @@ void PGMonitor::get_health(list<pair<health_status_t,string> >& summary,
...
+      float usage = ((float)p->second.kb_used) / ((float)p->second.kb);
+      if (usage > max_osd_usage)
+        max_osd_usage = usage;
+      if (usage < min_osd_usage)
+        min_osd_usage = usage;
+    }
+    float diff = max_osd_usage - min_osd_usage;
+    if (diff > g_conf->mon_warn_osd_usage_min_max_delta) {
+      ostringstream ss;
+      ss << "difference between min (" << roundf(min_osd_usage*1000.0)/100.0
```
Comment (dzafman) [inline src/mon/PGMonitor.cc]: There was follow on fix for this generated output.

**Comment by smithfarm** [comment]
@dzafman This passed another rados suite at http://tracker.ceph.com/issues/19009#note-71 and I added the commit you asked for. Please take a look.

**Pull request merged**